bootloaders/riotboot_dfu/main.c

```diff
@@ -36,9 +36,18 @@
 
 static bool _bootloader_alternative_mode(void)
 {
-#ifdef BTN_BOOTLOADER_PIN
+    bool state;
```

Comment:
You want to move that inside the condition below, otherwise the compiler will complain about the unused variable.